runtime/doc/dev_style.txt

```diff
@@ -41,6 +41,43 @@ All header files should start with `#pragma once` to prevent multiple inclusion.
         #pragma once
 <
 
+Headers system ~
+
+Nvim has uses two types of headers. There are "normal" headers and "defs"
+headers. Typically, each normal header will have a corresponding defs header,
+e.g. `fileio.h` and `fileio_defs.h`. This distinction is done to minimize
+recompilation on change. The goal is to achieve the following setup:
+
+- All headers (defs and normal) are allowed to include only defs headers, system
+headers and generated declarations. In other words, headers should never
```

Comment:
```suggestion
headers, and generated declarations. In other words, headers should never
```

Reply:
"should not" or "must not" (depending on whether exceptions can be made).